File: product-function/src/main/java/org/minhht/product/service/KafkaConsumerService.java
package org.minhht.product.service;

import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
public class KafkaConsumerService {
    @KafkaListener(topics = "product-topic", groupId = "product-group")
    public void consume(String message) {
        System.out.println("Product Service Rêcived: " + message);
    }
}

File: UserService/Service/KafkaConsumerService.cs
using Confluent.Kafka;
using System;
using System.Threading;

public class KafkaConsumerService
{
    public void ConsumeUserProfile()
    {
        var config = new ConsumerConfig
        {
            BootstrapServers = "localhost:9092",
            GroupId = "user-group",
            AutoOffsetReset = AutoOffsetReset.Earliest
        };

        using var consumer = new ConsumerBuilder<Null, string>(config).Build();
        consumer.Subscribe("user-profile-topic");

        while (true)
        {
            var result = consumer.Consume(CancellationToken.None);
            Console.WriteLine($"C# Consumer Received: {result.Message.Value}");
        }
    }
}